src/hooks/viewStateFacet/useGetEmbalmerCanClean.ts

```diff
@@ -0,0 +1,43 @@
+import { ViewStateFacet__factory } from '@sarcophagus-org/sarcophagus-v2-contracts';
+import { BigNumber } from 'ethers';
+import { useNetworkConfig } from 'lib/config';
+import { Sarcophagus } from 'types';
+import { useAccount, useContractRead } from 'wagmi';
+import { useGetGracePeriod } from './useGetGracePeriod';
+
+/**
+ * Uses `embalmerClaimWindow` from the contracts to check if the connected account can clean the
+ * sarcophagus. Returns `false` if the connected account is not the embalmer of the sarcophagus.
+ */
+export function useGetEmbalmerCanClean(sarcophagus: Sarcophagus | undefined): boolean {
+  const networkConfig = useNetworkConfig();
+  const gracePeriod = useGetGracePeriod();
+
+  const { data } = useContractRead({
+    address: networkConfig.diamondDeployAddress,
+    abi: ViewStateFacet__factory.abi,
+    functionName: 'getEmbalmerClaimWindow',
+  });
+
+  const { address } = useAccount();
+
+  if (!data) return false;
+  if (!sarcophagus || sarcophagus.isCleaned) return false;
```

Comment:
Do we need to handle if the sarco is buried? Or would that already be handled somewhere else?

Reply:
Looks like `hasLockedBond` is dynamically set in the viewstate facet, and will be false if the sarco is compromised, cleaned or buried. Which kinda removes the need to check `isCleaned` here too.

Thoughts? Do I keep `isCleaned` check here and add another for if buried, or rely on `hasLockedBond`, which does those checks?

Context contract code:
```
if (
                !sarcophagus
                    .cursedArchaeologists[sarcophagus.cursedArchaeologistAddresses[i]]
                    .isAccused &&
                    !sarcophagus.isCompromised &&
                    !sarcophagus.isCleaned &&
                    sarcophagus.resurrectionTime != 2 ** 256 - 1
            ) {
                // if the sarcophagus is not compromised, buried, or cleaned and
                // one or more unaccused archaeologists hasn't published a key share there is locked bond on the sarcophagus
                hasLockedBond = true;
            }
```

Reply:
Leaning toward relying on the contracts. Then I can do this:
`if (!sarcophagus || !sarcophagus.hasLockedBond) return false;` 
in addition to removing the `isEmbalmerCleanable` line completely